### ISSUE TITLE
[ScrollableAdapter] NPE doing Ant build

### ISSUE THREAD

**Issue by DSoa** [fappel/xiliary#47]
Installed CleanSheet today and enabled the Dark [Flat-Scrollbar] theme.  Ran an Ant build and got a bunch of errors.  The console view came up with a white background, also.  This is installed on Windows 10.  I went back to the default (non-CleanSheet) theme and restarted, but the errors still happen.  I'm reverting the install.  Looks promising, but maybe after it's been baked a little longer...

[Update] Unfortunately reverting the install did not take care of the error.  Any guidance of how to fix this would be appreciated.

!ENTRY org.eclipse.ui 4 4 2015-11-19 13:28:41.382
!MESSAGE An internal error has occurred.
!STACK 0
java.lang.NullPointerException
	at org.eclipse.ui.internal.WorkbenchPage.busyShowView(WorkbenchPage.java:1270)
	at org.eclipse.ui.internal.WorkbenchPage$12.run(WorkbenchPage.java:4211)
	at org.eclipse.swt.custom.BusyIndicator.showWhile(BusyIndicator.java:70)
	at org.eclipse.ui.internal.WorkbenchPage.showView(WorkbenchPage.java:4207)
	at org.eclipse.ui.internal.console.ConsoleManager$ShowConsoleViewJob.runInUIThread(ConsoleManager.java:321)
	at org.eclipse.ui.progress.UIJob$1.run(UIJob.java:97)
	at org.eclipse.swt.widgets.RunnableLock.run(RunnableLock.java:35)
	at org.eclipse.swt.widgets.Synchronizer.runAsyncMessages(Synchronizer.java:135)
	at org.eclipse.swt.widgets.Display.runAsyncMessages(Display.java:4155)
	at org.eclipse.swt.widgets.Display.readAndDispatch(Display.java:3772)
	at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine$4.run(PartRenderingEngine.java:1127)
	at org.eclipse.core.databinding.observable.Realm.runWithDefault(Realm.java:337)
	at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine.run(PartRenderingEngine.java:1018)
	at org.eclipse.e4.ui.internal.workbench.E4Workbench.createAndRunUI(E4Workbench.java:156)
	at org.eclipse.ui.internal.Workbench$5.run(Workbench.java:654)
	at org.eclipse.core.databinding.observable.Realm.runWithDefault(Realm.java:337)
	at org.eclipse.ui.internal.Workbench.createAndRunWorkbench(Workbench.java:598)
	at org.eclipse.ui.PlatformUI.createAndRunWorkbench(PlatformUI.java:150)
	at org.eclipse.ui.internal.ide.application.IDEApplication.start(IDEApplication.java:139)
	at org.eclipse.equinox.internal.app.EclipseAppHandle.run(EclipseAppHandle.java:196)
	at org.eclipse.core.runtime.internal.adaptor.EclipseAppLauncher.runApplication(EclipseAppLauncher.java:134)
	at org.eclipse.core.runtime.internal.adaptor.EclipseAppLauncher.start(EclipseAppLauncher.java:104)
	at org.eclipse.core.runtime.adaptor.EclipseStarter.run(EclipseStarter.java:380)
	at org.eclipse.core.runtime.adaptor.EclipseStarter.run(EclipseStarter.java:235)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
	at java.lang.reflect.Method.invoke(Unknown Source)
	at org.eclipse.equinox.launcher.Main.invokeFramework(Main.java:669)
	at org.eclipse.equinox.launcher.Main.basicRun(Main.java:608)
	at org.eclipse.equinox.launcher.Main.run(Main.java:1515)

**Comment by fappel** [comment]
Oh, I'm very sorry to hear that, but at the moment I have no clue what causes the persistent nature of this issue. AFAIK clean sheet should not persist any state besides some color settings for the syntax highlighting, which makes me wonder how this can happen.

Of course, I will have a look at it ASAP, but, unfortunately, I'm pretty busy today. So, it might take a day or two before I come up with a solution. In the meanwhile, you might check if reverting the install has really worked out successfully and remove the installed plugin and feature manually as mentioned in #32 if not.

The problem with the stack trace is that it seems as if the NPE isn't the root cause of the problem. Are there any other related traces available? You can delete all error log entries and execute the actions which lead to this problem. Afterward look at all traces that occurred - and post them here, if there are any.

**Comment by fappel** [comment]
On a second thought: Since you've mentioned that you use the dark theme extension with flat scrollbars, I wonder if you're using usually the original dark theme? I ask this just to ensure that it's not an inherited problem.

**Comment by DSoa** [comment]
This was a pretty new install of Eclipse and I just tried the Dark theme before seeing yours.  I hadn't tried an Ant build with just the Dark theme, so it is possible there could be a problem there.  I wasn't able to resolve it by reverting/uninstalling the plugin.  I went back to a fresh install now.  Sorry I didn't have any other logs.

**Comment by jbuchberger** [comment]
Looks a bit like https://bugs.eclipse.org/bugs/show_bug.cgi?id=471782
maybe the problems are somewhat related.

**Comment by fappel** [comment]
I was able to reproduce the layout problem, but not the NPE. This might be due to fixes that have been introduced in the meanwhile and it also might be related to https://bugs.eclipse.org/bugs/show_bug.cgi?id=471782 - I simply don't know as I wasn't able to reproduce the issue with a simple ant script.

However, I changed the dark theme approach in general. It is now a contribution to the original dark theme rather than being an extension. This solves the layout problems mentioned with respect to the console, for example, and eases maintenance as it avoids theme definition redundances.
